sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -172,9 +174,15 @@ private ConfidentialClientApplication getConfidentialClientApplication() {
                 if (certificatePassword == null) {
                     byte[] pemCertificateBytes = Files.readAllBytes(Paths.get(certificatePath));
 
-                    credential = ClientCredentialFactory.createFromCertificate(
-                        CertificateUtil.privateKeyFromPem(pemCertificateBytes),
-                        CertificateUtil.publicKeyFromPem(pemCertificateBytes));
+                    List<X509Certificate> x509CertificateList =  CertificateUtil.publicKeyFromPem(pemCertificateBytes);
+                    PrivateKey privateKey = CertificateUtil.privateKeyFromPem(pemCertificateBytes);
+                    if (x509CertificateList.size() == 1) {
+                        credential = ClientCredentialFactory.createFromCertificate(
+                            privateKey, x509CertificateList.get(0));
+                    } else {
+                        credential = ClientCredentialFactory.createFromCertificateChain(
+                            privateKey, x509CertificateList);
+                    }
```

Comment:
Do we need to special-case this for a list with size 1? Can `createFromCertificateChain` handle a list that has just 1 entry?

Reply:
yeah, we need to special case it, to ensure the cert chain is empty and not sent as part of the request, when there's only one cert to send.